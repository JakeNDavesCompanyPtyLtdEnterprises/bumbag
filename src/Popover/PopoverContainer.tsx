import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
// @ts-ignore
import ReakitPopover from 'reakit/Popover';

export interface LocalPopoverContainerProps {
  children: ((
    {
      isVisible,
      hide,
      show,
      toggle,
      ...args
    }: {
      isVisible: boolean;
      hide(): void;
      show(): void;
      toggle(): void;
    }
  ) => React.ReactNode);
  defaultVisible?: boolean;
  initialState?: {};
}
export type PopoverContainerProps = LocalPopoverContainerProps;

export const PopoverContainer: React.FunctionComponent<LocalPopoverContainerProps> = ({
  children,
  defaultVisible,
  initialState,
  ...props
}) => (
  <ReakitPopover.Container initialState={{ visible: defaultVisible, ...initialState }} {...props}>
    {({ visible, ...rest }: { visible: boolean }) => children({ isVisible: visible, ...rest } as any)}
  </ReakitPopover.Container>
);

PopoverContainer.propTypes = {
  children: PropTypes.func.isRequired,
  defaultVisible: PropTypes.bool,
  initialState: PropTypes.object
};
PopoverContainer.defaultProps = {
  defaultVisible: false,
  initialState: {}
};

export default PopoverContainer;
