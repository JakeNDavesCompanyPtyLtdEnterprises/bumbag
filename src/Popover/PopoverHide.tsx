import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { PopoverHideProps as ReakitPopoverHideProps } from 'reakit/ts';

import { PopoverHide as _PopoverHide } from './styled';

export interface LocalPopoverHideProps {
  children: React.ReactNode;
  hide(): void;
  onClick?(): void;
}
export type PopoverHideProps = ReakitPopoverHideProps & LocalPopoverHideProps;

export const PopoverHide: React.FunctionComponent<LocalPopoverHideProps> = ({ children, ...props }) => (
  <_PopoverHide {...props}>{children}</_PopoverHide>
);

PopoverHide.propTypes = {
  children: PropTypes.node.isRequired,
  hide: PropTypes.func.isRequired,
  onClick: PropTypes.func
};
PopoverHide.defaultProps = {
  onClick: undefined
};

const C: React.FunctionComponent<PopoverHideProps> = PopoverHide;
export default C;
