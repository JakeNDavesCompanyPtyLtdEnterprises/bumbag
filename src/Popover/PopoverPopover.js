// @flow
import React, { type Node } from 'react';

import TrapFocus from '../_utils/TrapFocus.tsx';
import { Box } from '../primitives';
import { getUniqueId } from '../uniqueId';
import _Popover from './styled';

type Props = {
  use?: any,
  children: Function | Node,
  className?: string,
  /** Delay of the animation if one is specified. */
  delay?: string,
  /** Duration of the animation if one is specified. */
  duration?: string,
  /** Will the popover have an expand animation when it is toggled on/off? */
  expand?: boolean | string,
  /** Should the popover be hidden when 'esc' is pressed?  */
  hideOnEsc?: boolean,
  /** Should the popover be hidden when outside is clicked?  */
  hideOnClickOutside?: boolean,
  /** Whether or not to show the popover component */
  isVisible?: boolean,
  /** Will the popover have a fade animation when it is toggled on/off? */
  fade?: boolean,
  flip?: boolean,
  gutter?: number | string,
  placement?:
    | 'auto'
    | 'top'
    | 'right'
    | 'bottom'
    | 'left'
    | 'top-start'
    | 'right-start'
    | 'bottom-start'
    | 'left-start'
    | 'top-end'
    | 'right-end'
    | 'bottom-end'
    | 'left-end',
  shift?: boolean,
  /** Will the popover have a slide animation when it is toggled on/off? */
  slide?: boolean | string,
  /** Timing of the animation */
  timing?: string,
  /** Traps focus within the popover */
  trapFocus?: boolean
};

const Popover = ({ children, isVisible, trapFocus, ...props }: Props) => (
  <_Popover visible={isVisible} {...props}>
    {trapFocus ? (
      <TrapFocus isActive={isVisible}>
        {({ fallbackFocusRef, initialFocusRef }) => (
          <Box elementRef={fallbackFocusRef}>
            {typeof children === 'function' ? children({ initialFocusRef }) : children}
          </Box>
        )}
      </TrapFocus>
    ) : typeof children === 'function' ? (
      children({})
    ) : (
      children
    )}
  </_Popover>
);

Popover.defaultProps = {
  animated: false,
  use: undefined,
  className: undefined,
  delay: undefined,
  duration: '250ms',
  expand: undefined,
  fade: false,
  flip: true,
  gutter: 12,
  hideOnEsc: true,
  hideOnClickOutside: true,
  isVisible: false,
  placement: 'bottom',
  popoverId: getUniqueId('Popover'),
  shift: true,
  slide: false,
  timing: 'ease-in-out',
  trapFocus: false
};

export default Popover;
