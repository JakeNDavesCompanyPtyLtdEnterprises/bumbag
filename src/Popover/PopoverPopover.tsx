import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { PopoverProps as ReakitPopoverProps } from 'reakit/ts';

import { isFunction } from '../_utils/assert';
import TrapFocus from '../_utils/TrapFocus';
import { Box } from '../primitives';
import {
  AnimateProps,
  Placement,
  RestrictHideProps,
  animateDefaultProps,
  placementPropType,
  restrictDefaultProps,
  animatePropTypes,
  restrictHidePropTypes
} from '../types';
import _Popover from './styled';

export interface Props {
  children: React.ReactNode | (({ initialFocusRef }: { initialFocusRef?: React.RefObject<any> }) => React.ReactNode);
  className?: string;
  /** Whether or not to show the popover component */
  isVisible?: boolean;
  flip?: boolean;
  gutter?: number | string;
  placement?: Placement;
  shift?: boolean;
  /** Traps focus within the popover */
  trapFocus?: boolean;
}
export type LocalPopoverPopoverProps = Props & AnimateProps & RestrictHideProps;
export type PopoverPopoverProps = LocalPopoverPopoverProps & ReakitPopoverProps;

export const PopoverPopover: React.FunctionComponent<LocalPopoverPopoverProps> = ({
  children,
  isVisible,
  trapFocus,
  ...props
}) => (
  <_Popover visible={isVisible} {...props}>
    {trapFocus ? (
      <TrapFocus isActive={isVisible}>
        {({ fallbackFocusRef, initialFocusRef }) => (
          <Box elementRef={fallbackFocusRef}>{isFunction(children) ? children({ initialFocusRef }) : children}</Box>
        )}
      </TrapFocus>
    ) : isFunction(children) ? (
      children({})
    ) : (
      children
    )}
  </_Popover>
);

export const popoverPopoverPropTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  className: PropTypes.string,
  isVisible: PropTypes.bool,
  flip: PropTypes.bool,
  gutter: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  placement: placementPropType,
  shift: PropTypes.bool,
  trapFocus: PropTypes.bool,
  ...animatePropTypes,
  ...restrictHidePropTypes
};
PopoverPopover.propTypes = popoverPopoverPropTypes;

export const popoverPopoverDefaultProps: Partial<LocalPopoverPopoverProps> = {
  className: undefined,
  flip: true,
  gutter: 12,
  isVisible: false,
  placement: 'bottom',
  shift: true,
  trapFocus: false,
  ...animateDefaultProps,
  ...restrictDefaultProps,
  hideOnEsc: true,
  hideOnClickOutside: true
};
PopoverPopover.defaultProps = popoverPopoverDefaultProps;

const C: React.FunctionComponent<PopoverPopoverProps> = PopoverPopover;
export default C;
