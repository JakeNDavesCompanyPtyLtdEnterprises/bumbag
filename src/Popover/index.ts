export { default } from './Popover';
export { default as Popover } from './Popover';
export { default as PopoverContainer } from './PopoverContainer';
export { default as PopoverPopover } from './PopoverPopover';
export { default as PopoverShow } from './PopoverShow';
export { default as PopoverHide } from './PopoverHide';
export { default as PopoverToggle } from './PopoverToggle';
export { default as PopoverClose } from './PopoverClose';
