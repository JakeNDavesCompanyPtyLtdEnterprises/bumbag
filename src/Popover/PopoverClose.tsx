import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { ButtonProps } from '../Button/Button';

import { PopoverClose as _PopoverClose } from './styled';
import PopoverHide from './PopoverHide';
import Icon from '../Icon';
import VisuallyHidden from '../VisuallyHidden';
import { Omit } from '../types';

export interface LocalPopoverCloseProps {
  hide(): void;
}
export type PopoverCloseProps = Omit<ButtonProps, 'children'> & LocalPopoverCloseProps;

export const PopoverClose: React.FunctionComponent<LocalPopoverCloseProps> = props => (
  <_PopoverClose use={PopoverHide} size="small" kind="link" {...props}>
    <VisuallyHidden>Close</VisuallyHidden>
    <Icon a11yLabel="Close" aria-hidden="true" icon="times" />
  </_PopoverClose>
);

PopoverClose.propTypes = {
  hide: PropTypes.func.isRequired
};
PopoverClose.defaultProps = {};

// @ts-ignore
const C: React.FunctionComponent<PopoverCloseProps> = PopoverClose;
export default C;
