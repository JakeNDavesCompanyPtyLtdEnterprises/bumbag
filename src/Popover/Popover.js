// @flow
import React, { Fragment, type Element } from 'react';

import type { Placement } from '../typesold';
import { InlineBlock } from '../primitives';
import { getUniqueId } from '../uniqueId';
import Pane from '../Pane';
import PopoverContainer from './PopoverContainer';
import PopoverClose from './PopoverClose';
import PopoverPopover from './PopoverPopover';
import PopoverShow from './PopoverShow';
import PopoverHide from './PopoverHide';
import PopoverToggle from './PopoverToggle';

type Props = {
  use?: any,
  children: Element<any>,
  className?: string,
  content: string | Element<any> | Function,
  /** Delay of the animation if one is specified. */
  delay?: string,
  /** Duration of the animation if one is specified. */
  duration?: string,
  /** Will the popover have an expand animation when it is toggled on/off? */
  expand?: boolean | string,
  /** Should the popover be hidden when 'esc' is pressed?  */
  hideOnEsc?: boolean,
  /** Should the popover be hidden when outside is clicked?  */
  hideOnClickOutside?: boolean,
  /** Whether or not to show the popover component */
  isVisible?: boolean,
  /** Will the popover have a fade animation when it is toggled on/off? */
  fade?: boolean,
  flip?: boolean,
  gutter?: number | string,
  /** Placement of the popover content */
  placement?: Placement,
  shift?: boolean,
  /** Displays a cross button in the top right corner of the popover content. */
  showCloseButton?: boolean,
  /** Will the popover have a slide animation when it is toggled on/off? */
  slide?: boolean | string,
  /** Timing of the animation */
  timing?: string,
  /** Traps focus within the popover */
  trapFocus?: boolean
};

const Popover = ({ children, content, showCloseButton, ...props }: Props) => (
  <PopoverContainer>
    {popover => (
      <InlineBlock relative>
        {React.cloneElement(children, { use: PopoverToggle, ...popover })}
        <PopoverPopover border="shadow" padding="small" {...props} {...popover} use={Pane}>
          {({ initialFocusRef }) => (
            <Fragment>
              {showCloseButton && <PopoverClose elementRef={initialFocusRef} {...popover} />}{/* eslint-disable-line */}
              {typeof content === 'function' ? content({ initialFocusRef, ...popover }) : content}
            </Fragment>
          )}
        </PopoverPopover>
      </InlineBlock>
    )}
  </PopoverContainer>
);

Popover.Popover = PopoverPopover;
Popover.Container = PopoverContainer;
Popover.Close = PopoverClose;
Popover.Hide = PopoverHide;
Popover.Show = PopoverShow;
Popover.Toggle = PopoverToggle;

Popover.defaultProps = {
  animated: false,
  use: undefined,
  className: undefined,
  delay: undefined,
  duration: '100ms',
  expand: undefined,
  fade: false,
  flip: true,
  gutter: 12,
  hideOnEsc: true,
  hideOnClickOutside: true,
  isVisible: false,
  placement: 'bottom',
  popoverId: getUniqueId('Popover'),
  shift: true,
  showCloseButton: false,
  slide: false,
  timing: 'ease-in-out',
  trapFocus: false
};

export default Popover;
