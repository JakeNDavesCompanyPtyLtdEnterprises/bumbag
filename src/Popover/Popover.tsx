import * as React from 'react';
import * as PropTypes from 'prop-types';

import { InlineBlock } from '../primitives';
import Pane from '../Pane';
import PopoverContainer, { PopoverContainerProps } from './PopoverContainer';
import PopoverClose, { PopoverCloseProps } from './PopoverClose';
import PopoverPopover, {
  LocalPopoverPopoverProps,
  PopoverPopoverProps,
  popoverPopoverDefaultProps,
  popoverPopoverPropTypes
} from './PopoverPopover';
import PopoverShow, { PopoverShowProps } from './PopoverShow';
import PopoverHide, { PopoverHideProps } from './PopoverHide';
import PopoverToggle, { PopoverToggleProps } from './PopoverToggle';

export type LocalPopoverProps = LocalPopoverPopoverProps & {
  children: React.ReactElement<any>;
  className?: string;
  content:
    | string
    | React.ReactElement<any>
    | (({ initialFocusRef, ...args }: { initialFocusRef?: React.RefObject<any> }) => React.ReactNode);
  /** Displays a cross button in the top right corner of the popover content. */
  showCloseButton?: boolean;
};
export type PopoverProps = LocalPopoverProps;
export type PopoverComponents = {
  Popover: React.FunctionComponent<PopoverPopoverProps>;
  Container: React.FunctionComponent<PopoverContainerProps>;
  Close: React.FunctionComponent<PopoverCloseProps>;
  Hide: React.FunctionComponent<PopoverHideProps>;
  Show: React.FunctionComponent<PopoverShowProps>;
  Toggle: React.FunctionComponent<PopoverToggleProps>;
};

export const Popover: React.FunctionComponent<LocalPopoverProps> & PopoverComponents = ({
  children,
  content,
  showCloseButton,
  ...props
}) => (
  <PopoverContainer>
    {popover => (
      <InlineBlock relative>
        {React.cloneElement(children, { use: PopoverToggle, ...popover })}
        <PopoverPopover elevation="200" padding="minor-5" {...props} {...popover} use={Pane}>
          {({ initialFocusRef }) => (
            <React.Fragment>
              {showCloseButton && <PopoverClose elementRef={initialFocusRef} {...popover} />}{/* eslint-disable-line */}
              {typeof content === 'function' ? content({ initialFocusRef, ...popover }) : content}
            </React.Fragment>
          )}
        </PopoverPopover>
      </InlineBlock>
    )}
  </PopoverContainer>
);

Popover.Popover = PopoverPopover;
Popover.Container = PopoverContainer;
Popover.Close = PopoverClose;
Popover.Hide = PopoverHide;
Popover.Show = PopoverShow;
Popover.Toggle = PopoverToggle;

Popover.propTypes = {
  ...popoverPopoverPropTypes,
  className: PropTypes.string,
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.element, PropTypes.func]) as PropTypes.Validator<
    LocalPopoverProps['content']
  >,
  showCloseButton: PropTypes.bool,
  children: PropTypes.element.isRequired
};
Popover.defaultProps = {
  ...popoverPopoverDefaultProps,
  className: undefined,
  showCloseButton: false,
  children: undefined
};

// @ts-ignore
const C: React.FunctionComponent<PopoverProps> & PopoverComponents = Popover;
export default C;
