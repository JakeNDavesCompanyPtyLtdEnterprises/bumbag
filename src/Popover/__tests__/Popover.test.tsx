import * as React from 'react';
import render from '../../_utils/tests/render';
import Button from '../../Button';
// @ts-ignore
import Text from '../../Text';
import Popover from '../Popover';

it('renders correctly for a default popover', () => {
  const { container } = render(
    <Popover content={<Text>Hello world</Text>}>
      <Button>Toggle popover</Button>
    </Popover>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when a placement prop is set', () => {
  const { container } = render(
    <Popover content={<Text>Hello world</Text>} placement="right">
      <Button>Toggle popover</Button>
    </Popover>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly when showCloseButton is truthy', () => {
  const { container } = render(
    <Popover content={<Text>Hello world</Text>} showCloseButton>
      <Button>Toggle popover</Button>
    </Popover>
  );
  expect(container.firstChild).toMatchSnapshot();
});
