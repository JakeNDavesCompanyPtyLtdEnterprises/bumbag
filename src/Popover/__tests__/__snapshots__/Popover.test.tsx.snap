// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a default popover 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c1.c1.c1 {
  position: relative;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: white;
  color: inherit;
  padding: 1.25rem !important;
}

.c6:focus:not(:focus-visible) {
  outline: none;
}

.c9 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c9:focus:not(:focus-visible) {
  outline: none;
}

.c5 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c5[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 1px solid #b3b3b3;
}

.c2:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7c81d9 0px 0px 0px 2px;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2:hover {
  background-color: #efefef;
}

.c2:hover:active {
  background-color: #e3e3e3;
}

abbr.c8 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c8 {
  font-style: italic;
}

kbd.c8 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c8 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c8 {
  font-weight: bold;
}

sub.c8,
sup.c8 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c8,
u.c8 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c8,
cite.c8,
var.c8,
em.c8 {
  font-style: italic;
}

kbd.c8,
pre.c8,
samp.c8 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c8 .c10 {
  top: 0.15em;
}

.c7 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0px 2px 4px 0px #e6e6e6,0px 0px 0px 1px #e6e6e6;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  cursor: auto;
  z-index: 999;
  display: block;
}

<div
  class="c0 c1"
>
  <button
    aria-controls="popover0"
    aria-haspopup="true"
    class="c2 c3 "
    kind="default"
    type="button"
  >
    Toggle popover
  </button>
  <div
    aria-hidden="true"
    class="c4 c5 c6 c7"
    data-placement="bottom"
    hidden=""
    id="popover0"
    role="group"
  >
    <span
      class="c8 c9"
    >
      Hello world
    </span>
  </div>
</div>
`;

exports[`renders correctly when a placement prop is set 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c1.c1.c1 {
  position: relative;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: white;
  color: inherit;
  padding: 1.25rem !important;
}

.c6:focus:not(:focus-visible) {
  outline: none;
}

.c9 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c9:focus:not(:focus-visible) {
  outline: none;
}

.c5 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c5[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  display: inline-block;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 1px solid #b3b3b3;
}

.c2:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7c81d9 0px 0px 0px 2px;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2:hover {
  background-color: #efefef;
}

.c2:hover:active {
  background-color: #e3e3e3;
}

abbr.c8 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c8 {
  font-style: italic;
}

kbd.c8 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c8 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c8 {
  font-weight: bold;
}

sub.c8,
sup.c8 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c8,
u.c8 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c8,
cite.c8,
var.c8,
em.c8 {
  font-style: italic;
}

kbd.c8,
pre.c8,
samp.c8 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c8 .c10 {
  top: 0.15em;
}

.c7 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0px 2px 4px 0px #e6e6e6,0px 0px 0px 1px #e6e6e6;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  cursor: auto;
  z-index: 999;
  display: block;
}

<div
  class="c0 c1"
>
  <button
    aria-controls="popover1"
    aria-haspopup="true"
    class="c2 c3 "
    kind="default"
    type="button"
  >
    Toggle popover
  </button>
  <div
    aria-hidden="true"
    class="c4 c5 c6 c7"
    data-placement="right"
    hidden=""
    id="popover1"
    role="group"
  >
    <span
      class="c8 c9"
    >
      Hello world
    </span>
  </div>
</div>
`;

exports[`renders correctly when showCloseButton is truthy 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c1.c1.c1 {
  position: relative;
}

.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c3:focus:not(:focus-visible) {
  outline: none;
}

.c6 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: white;
  color: inherit;
  padding: 1.25rem !important;
}

.c6:focus:not(:focus-visible) {
  outline: none;
}

.c11 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c11:focus:not(:focus-visible) {
  outline: none;
}

.c5 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c5[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  display: inline-block;
}

.c13 {
  fill: inherit;
  position: relative;
  font-size: 1em;
  height: 1em;
  width: 1em;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 1px solid #b3b3b3;
}

.c2:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7c81d9 0px 0px 0px 2px;
}

.c2[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c2:hover {
  background-color: #efefef;
}

.c2:hover:active {
  background-color: #e3e3e3;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 1px solid #b3b3b3;
}

.c9:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7c81d9 0px 0px 0px 2px;
}

.c9[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c9 {
  border: 0;
  background: unset;
  color: #435a6f;
  fill: #435a6f;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover {
  color: #000;
  fill: #000;
}

abbr.c14 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c14 {
  font-style: italic;
}

kbd.c14 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c14 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c14 {
  font-weight: bold;
}

sub.c14,
sup.c14 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c14,
u.c14 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c14,
cite.c14,
var.c14,
em.c14 {
  font-style: italic;
}

kbd.c14,
pre.c14,
samp.c14 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c14 .c12 {
  top: 0.15em;
}

.c7 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-shadow: 0px 2px 4px 0px #e6e6e6,0px 0px 0px 1px #e6e6e6;
}

.c8 {
  position: absolute;
  width: 24px;
  height: 24px;
  right: 0;
  top: 0;
  padding: 0px;
  text-align: right;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  cursor: auto;
  z-index: 999;
  display: block;
}

.c10 {
  border: 0px;
  -webkit-clip: rect(0px,0px,0px,0px);
  clip: rect(0px,0px,0px,0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  position: absolute;
}

<div
  class="c0 c1"
>
  <button
    aria-controls="popover2"
    aria-haspopup="true"
    class="c2 c3 "
    kind="default"
    type="button"
  >
    Toggle popover
  </button>
  <div
    aria-hidden="true"
    class="c4 c5 c6 c7"
    data-placement="bottom"
    hidden=""
    id="popover2"
    role="group"
  >
    <button
      aria-controls="popover2"
      aria-haspopup="true"
      class="c8 c9 c3 "
      kind="link"
      type="button"
    >
      <div
        class="c10 c11"
      >
        Close
      </div>
      <svg
        aria-hidden="true"
        class="c12 c13 c11"
        role="img"
        size="1"
        style="size: 1px;"
        viewBox="0 0 352 512"
      >
        <title>
          Close
        </title>
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
          fill="currentColor"
        />
      </svg>
    </button>
    <span
      class="c14 c11"
    >
      Hello world
    </span>
  </div>
</div>
`;
